### ISSUE TITLE
Dask worker fails to open badly translated grib file with Iris

### ISSUE THREAD

**Issue by pelson** [SciTools/iris-grib#71]
First, the traceback:

```
  File "/data/pelson/conda/envs/data-transform/lib/python2.7/site-packages/iris_grib/_load_convert.py", line 1634, in generating_process
    if options.warn_on_unsupported:
AttributeError: 'thread._local' object has no attribute 'warn_on_unsupported'
```

----

There is some weird interaction between the pickled function and the way the function is called with dask. Essentially, I think iris-grib is relying on the _load_convert having been run, but that isn't necessarily the case when pickling/unpickling the function (or at least, the local it is referencing isn't the one that the thread has run the module on). This is a somewhat woolly description :stuck_out_tongue_winking_eye:.

Rather than depending on the module having been run, we either need to ensure that we validate the [threading.local](https://docs.python.org/2/library/threading.html#threading.local) has been correctly configured, or just ditch the complexity. I don't currently see the usecase for having warn_on_unsupported be defined on a per-thread basis. The implementation of this was originally done in https://github.com/SciTools/iris/commit/0248bf3abaf8d2ba77cc0c3b82cea23e87753ef5, and for all of those tests, I believe a module level constant would suffice.

**Comment by pelson** [comment]
Right, so I believe the reason it exists is so that we can change the value in our *multi-processing* unit-tests. As a result, I believe a module level constant will work just as well.

**Comment by pelson** [comment]
For my own sanity, I replicated the problem without dask or iris. It is simply a case that *of course* the ```threading.local``` object is not modified for all threads (that it its reason for existence), so we must setup ``options`` each time that we expect options to be defined (because we don't know what thread a function gets run from).

I have provided some simple code to reproduce the issue at https://github.com/SciTools/iris-grib/pull/72#issuecomment-265290218.
